the-collab-lab/tcl-25-smart-shopping-list
# Issue #6 Prevent duplicate entries and implement form validation

## Description

This PR prevents the user from adding duplicate entries to the shopping list. It compares db items with the ones submitted by the user via input and makes sure that items with the same name do not get added to the shopping list. If the user tries to submit a duplicate item, they are shown an error message

## Related Issue



## Acceptance Criteria

- [x] Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalisation has been normalised and punctuation has been removed

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin ku-ci-prevent-duplicates` and check that branch out with `git checkout ku-ci-prevent-duplicates` 
* Then `npm start` to launch the app
* Clear local storage. Generate a new token for setting up a shopping list. Submit “Add " form with no input to make sure an empty entry doesn't get added to the list
* Enter an item that is already on the list to make sure you're alerted that the item is already on the list. Use capitalisation and punctuation (f.ex  `To?m/a&t@oe$s%`). 
* Submit an item with no frequency selected to make sure you're alerted that this field is required. 
* Make sure duplicate items do not appear in list view and in the db
